File: library/src/main/java/me/proxer/library/entity/info/EpisodeInfo.java
package me.proxer.library.entity.info;

import com.squareup.moshi.Json;
import lombok.Value;
import me.proxer.library.enums.Category;
import me.proxer.library.enums.MediaLanguage;

import java.util.List;
import java.util.Set;

/**
 * Entity containing information about the available episodes or chapters of an media entry.
 *
 * @author Ruben Gees.
 */
@Value
public class EpisodeInfo {

    /**
     * Returns the first available episode.
     */
    @Json(name = "start")
    private int firstEpisode;

    /**
     * Returns the last available episode.
     */
    @Json(name = "end")
    private int lastEpisode;

    /**
     * Returns the category of the associated media entry.
     */
    @Json(name = "kat")
    private Category category;

    /**
     * Returns the available languages.
     */
    @Json(name = "lang")
    private Set<MediaLanguage> availableLanguages;

    /**
     * Returns the progress, the user has made on this media entry so far.
     */
    @Json(name = "state")
    private int userProgress;

    /**
     * Returns the actual list of episodes.
     */
    @Json(name = "episodes")
    private List<Episode> episodes;
}

File: library/src/main/kotlin/me/proxer/library/entity/info/EpisodeInfo.kt
package me.proxer.library.entity.info

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.enums.Category
import me.proxer.library.enums.MediaLanguage

/**
 * Entity containing information about the available episodes or chapters of an [Entry].
 *
 * @property firstEpisode The first available episode.
 * @property lastEpisode The last available episode.
 * @property category The category of the associated media entry.
 * @property availableLanguages The available languages.
 * @property userProgress The progress, the user has made on this media entry so far.
 * @property episodes The actual list of episodes.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class EpisodeInfo(
    @Json(name = "start") val firstEpisode: Int,
    @Json(name = "end") val lastEpisode: Int,
    @Json(name = "kat") val category: Category,
    @Json(name = "lang") val availableLanguages: Set<MediaLanguage>,
    @Json(name = "state") val userProgress: Int,
    @Json(name = "episodes") val episodes: List<Episode>
)
